### DISCUSSION TITLE
DPI Settings Options

### DISCUSSION THREAD

**Discussion by DevAnyKR** [vincerubinetti/svg-to-png#17] [Ideas]
custom DPI

96, 100, 200, 300, 600, etc...

**Comment by vincerubinetti** [comment]
So, in digital images, a pixel is just a pixel. The DPI is really just a "hint" to whatever is displaying it that it should result in a certain physical, real-world size: https://stackoverflow.com/a/27328679/2180570

This tool already allows you to "increase the dpi" (sort of) by just increasing the size of your output image, giving you more pixels and thus resolution.

I was unaware of this, but apparently PNG does have the ability to store DPI metadata. But I'm not seeing anything online for how to get the browser to change the dpi of the saved png when calling [toDataUrl](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/toDataURL) (what this tool relies on under-the-hood to do the conversion). It says there:

> The created image data will have a resolution of 96dpi for file formats that support encoding resolution metadata.

So, to support changing this in the tool, I'd have to manually manipulate the byte data in the PNG, which could add a lot of complexity and brittle-ness.

Probably best to just rely on the tool's sizing controls to render at a very high resolution, then use [image editing software](https://www.photopea.com/) to resize (and set DPI) as needed.
